src/framework/mpas_io.F

```diff
@@ -1984,22 +1993,40 @@ subroutine MPAS_io_get_var_generic(handle, fieldname, intVal, intArray1d, intArr
 !         call mpas_log_write('  value is char')
 
          charVal_p => charVal
+
+         ! get the dimension of the char variable to ensure the provided output buffer is large enough
+         call MPAS_io_inq_var(handle, fieldname, fieldtype, ndims, dimnames, dimsizes, local_ierr)
+         do i = 1, ndims
+            message = '  MPAS_io_get_var_generic len(charVal):$i var "'//trim(fieldname)// &
+                      '" type is $i dim $i is '// trim(dimnames(i))//' size is $i'
+            IO_DEBUG_WRITE(message , intArgs=(/len(charVal), fieldtype, i, dimsizes(i)/))
+         end do
+         ! because charVal is provided, assume dimension 1 is the string length
+         if (dimsizes(1) > len(charVal)) then
+            local_ierr = MPAS_IO_ERR_INSUFFICIENT_BUF
+            message = 'MPAS_io_get_var_generic var "'//trim(fieldname)//  &
+                      '" len too big, len=$i buflen=$i'
```

Comment:
It might make this message more helpful to users if we used something like:
```fortran
            message = 'Length of string variable "'//trim(fieldname)//'" in file "'//trim(handle % filename)//'"'
            IO_ERROR_WRITE(message, intArgs=(/0/))
            message = '    exceeds buffer size: len('//trim(fieldname)//')=$i, len(buffer)=$i'
            IO_ERROR_WRITE(message, intArgs=(/dimsizes(1), len(charVal)/)) 
```
This would generate an error log entry something like:
```
ERROR: Length of string variable "xtime" in file "x1.10242.init.nc"
ERROR:     exceeds buffer size: len(xtime)=128, len(buffer)=64
```